plugins/rbac-backend/docs/apis.md

```diff
@@ -0,0 +1,409 @@
+
+Deletes a permission policy of a specified entity.
+
+Request Parameters:
+
+| Parameter name | Description                                                                   | Type   |
+| -------------- | ----------------------------------------------------------------------------- | ------ |
+| kind           | Kind of the entity                                                            | String |
+| namespace      | Namespace of the entity                                                       | String |
+| name           | Username of the entity                                                        | String |
+| permission     | Permission from a specific plugin, Resource type or name                      | String |
+| policy         | Policy action for the permission, `create`, `read`, `update`, `delete`, `use` | String |
+| effect         | `allow` or `deny`                                                             | String |
+
+Returns a status code of 204 upon success.
+
+---
+
+## Role
+
+### GET role
+
+GET </api/permission/roles>
+
+lists all roles.
+
+Returns:
+
+```json
+[
+  {
+    "memberReferences": ["user:default/adam"],
+    "name": "role:default/guests"
+  },
+  {
+    "memberReferences": ["group:default/janus-authors", "user:default/matt"],
+    "name": "role:default/test"
+  }
+]
+```
+
+---
+
+GET </api/permission/roles/:kind/:namespace/:name>
+ex. <http://localhost:7007/api/permission/roles/role/default/test>
+
+List the single role and the members associated with that role.
+
+Request Parameters:
+
+| Parameter name | Description           | Type   |
+| -------------- | --------------------- | ------ |
+| kind           | Kind of the role      | String |
+| namespace      | Namespace of the role | String |
+| name           | name of the role      | String |
+
+Returns:
+
+```json
+[
+  {
+    "memberReferences": ["group:default/janus-authors", "user:default/matt"],
+    "name": "role:default/test"
+  }
+]
+```
+
+---
+
+### POST role
+
+POST </api/permission/roles>
+
+Creates a new role.
+
+Request Parameters:
+
+| Parameter name   | Description                                                      | Type   |
+| ---------------- | ---------------------------------------------------------------- | ------ |
+| memberReferences | users / groups to be added to the role `<kind>:<default>/<name>` | Array  |
+| name             | name of the role                                                 | String |
+
+body:
+
+```json
+{
+  "memberReferences": ["group:default/test"],
+  "name": "role:default/test_admin"
+}
+```
+
+Returns a status code of 201 upon success.
+
+---
+
+### PUT role
+
+PUT </api/permission/roles/:kind/:namespace/:name>
+ex. <http://localhost:7007/api/permission/roles/role/default/test_admin>
+
+Updates a specified role.
+
+Request Parameters:
+
+| Parameter name | Description           | Type   |
+| -------------- | --------------------- | ------ |
+| kind           | Kind of the role      | String |
```

Comment:
just for my understanding, what does kind of role mean here? or possible values?

will it be `role`, `user` , `group`

Reply:
Updated it to now have just `role` to hopefully reduce confusion.